ibis/backends/dask/execution/util.py

```diff
@@ -345,3 +348,116 @@ def safe_scalar_type(output_meta):
         output_meta = pd.Timestamp(1, tz=output_meta.tz, unit=output_meta.unit)
 
     return output_meta
+
+
+def add_partitioned_sorted_column(
+    df: Union[dd.DataFrame, dd.Series],
+    set_index: bool = False,
+    col_name: str = "_ibis_index",
+) -> Tuple[dd.DataFrame, List]:
+    """Add a column that is already partitioned and sorted
+    This columns acts as if we had a global index across the distributed data.
+    Important properties:
+    - Each row has a unique id (i.e. a value in this column)
+    - IDs within each partition are already sorted
+    - Any id in partition N_{t} is less than any id in partition N_{t+1}
+    We do this by designating a sufficiently large space of integers per
```

Comment:
What is the `sufficiently large space` here?

Reply:
I kind of like the Spark way of describing this. pretty clear to me what's the max size of records in a partition and what is the max number of partitions and what is the size the the id column.

https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.sql.functions.monotonically_increasing_id.html#pyspark.sql.functions.monotonically_increasing_id

Reply:
I can add specific text here (or a user can do the `x << 32` math here, but for all practical purposes it shouldn't matter and you should never need to know/rely on the specifics of "sufficiently large".

Reply:
This is fine leaving unchanged.

My point is mainly that we never know how user would break our code, e.g., if I managed to put 2^33 rows inside one partition this would break :(

Reply:
> On the side, perhaps we should validate inside each partition that we don't overflow.

I think the only way to do that would be to actually compute the lengths in each partition?

Reply:
```
    def helper(
        df: Union[dd.Series, dd.DataFrame],
        partition_info: Dict[str, Any],  # automatically injected by dask
        col_name: str,
    ):
```

I think inside each of these we already have the data for each partition so we can check if that overflows?

Reply:
Ah fair - I can throw an error if the len here is larger than the allowable number. This will only trigger at compute time.

Reply:
Computation time is fine - better than nothing XD